File: answers/medium/2294. Partition Array Such That Maximum Difference Is K.go
import (
	"sort"
)

func partitionArray(nums []int, k int) int {
    sort.Ints(nums)
    sp := 1;
    prev := nums[0];
    for i := 1; i < len(nums); i++ {
        if nums[i] - prev > k {
            sp++;
            prev = nums[i];
        }
    }
    return sp;
}

File: answers/medium/2294. Partition Array Such That Maximum Difference Is K.py
class Solution:
    def partitionArray(self, nums: List[int], k: int) -> int:
        nums.sort()
        sp = 1
        prev = nums[0]
        for i in range(len(nums)):
            if nums[i] - prev > k:
                sp += 1
                prev = nums[i]
        return sp